### PR TITLE
Open shift project environment created event

### PR DESCRIPTION
**Pull request by andrejonathan**
### Description
This PR introduces a queryable event handler for the OpenShiftProjectEnvironmentCreatedEvent and adds a new field

### Essential Checks:

* [NO] Have you added tests where necessary?
* [NA] Have you added metric gathering code where necessary (in `EventHandlers` and `CommandHandlers`)?
* [NA] Have you added new commands to the displayable Help list?
* [NA] Have you updated any necessary documentation?

### CODE REVIEW

File: src/gluon/events/team/TeamOpenShiftCloudMigrated.ts [hunk 1]
```diff
@@ -209,7 +210,7 @@ export class TeamOpenShiftCloudMigrated extends BaseQMEvent implements HandleEve
         const openShiftNonProd: OpenShiftConfig = QMConfig.subatomic.openshiftClouds[team.openShiftCloud].openshiftNonProd;
 
         taskRunner.addTask(
-            new CreateOpenshiftEnvironments(createOpenshiftEnvironmentsDetails, environmentsForCreation, openShiftNonProd),
+            new CreateOpenshiftEnvironments(new AtomistQMContext(ctx), createOpenshiftEnvironmentsDetails, environmentsForCreation, openShiftNonProd, team),
```
Comment (kbristow) [inline src/gluon/events/team/TeamOpenShiftCloudMigrated.ts]: I think instead of passing the HandlerContext down the various function calls, please pass the AtomistQMContext through instead. Use the QMContext as the parameter type though.

File: src/gluon/tasks/project/CreateOpenshiftEnvironments.ts [hunk 2]
```diff
@@ -60,13 +63,32 @@ export class CreateOpenshiftEnvironments extends Task {
...
+                owningTenant: this.environmentsRequestedEvent.owningTenant,
+                project: this.environmentsRequestedEvent.project,
+                displayName: environment.displayName,
+                postfix: environment.postfix,
+            };
+
+            logger.debug(`openShiftProjectEnvironmentCreatedEvent: ${JSON.stringify(openShiftProjectEnvironmentCreatedEvent)}`);
+
+            await this.ctx.raiseEvent(openShiftProjectEnvironmentCreatedEvent, "OpenShiftProjectEnvironmentCreatedEvent");
+
```
Comment (kbristow) [inline src/gluon/tasks/project/CreateOpenshiftEnvironments.ts]: Can we move this to a function for clarity? 
```javascript
this.createOpenshiftProject(...);
this.raiseOpenshiftEnvironmentCreatedEvent(...);
```
would be a nicer read I think?

File: src/graphql/ingester/GluonTeam.graphql [hunk 3]
```diff
@@ -17,3 +18,13 @@ type Member {
 	domainUsername: String
 	slackIdentity: SlackIdentity
 }
+
+type MetaData {
```
Comment (kbristow) [inline src/graphql/ingester/GluonTeam.graphql]: Please move the MetaData types into their own Ingester files. They will not only be used on Teams, so a neutral file makes more sense.

Comment (andrejonathan) [reply src/graphql/ingester/GluonTeam.graphql]: agreed, done

File: src/gluon/tasks/project/CreateOpenshiftEnvironments.ts (hunk 2 above)
Comment (andrejonathan) [reply src/gluon/tasks/project/CreateOpenshiftEnvironments.ts]: yes, done

File: src/gluon/events/team/TeamOpenShiftCloudMigrated.ts (hunk 1 above)
Comment (andrejonathan) [reply src/gluon/events/team/TeamOpenShiftCloudMigrated.ts]: So still new it up but outside and then pass it in?

Comment (andrejonathan) [reply src/gluon/events/team/TeamOpenShiftCloudMigrated.ts]:
```            const atomistQMContext: QMContext = new AtomistQMContext(ctx);
            taskRunner.addTask(
                new CreateOpenshiftEnvironments(atomistQMContext, environmentsRequestedEvent, environmentsForCreation, openshiftNonProd, owningTeam),
            ).addTask(
                new ConfigureJenkinsForProject(environmentsRequestedEvent, project.devDeploymentPipeline, project.releaseDeploymentPipelines, openshiftNonProd),
            );
```
like this?

Comment (andrejonathan) [reply src/gluon/events/team/TeamOpenShiftCloudMigrated.ts]: if not im not sure what you mean

Comment (andrejonathan) [reply src/gluon/events/team/TeamOpenShiftCloudMigrated.ts]: and this is our abstraction of ctx, yes?

Comment (andrejonathan) [reply src/gluon/events/team/TeamOpenShiftCloudMigrated.ts]: our/your :)

Comment (kbristow) [reply src/gluon/events/team/TeamOpenShiftCloudMigrated.ts]: You swapped all the QMMessageClients to HandlerContexts. I am suggesting changing all the QMMessageClient usages to QMContext usages instead.

Comment (andrejonathan) [reply src/gluon/events/team/TeamOpenShiftCloudMigrated.ts]: Bilal must have, his branch, but I will take a look

File: src/graphql/ingester/OpenShiftProjectEnvironmentCreatedEvent.graphql [hunk 4]
```diff
@@ -0,0 +1,16 @@
+type OpenShiftProjectEnvironmentCreatedEvent @rootType {
+    owningTeam(name: String): GluonTeam
+    namespace: String
+    masterUrl: String
+    owningTenant: OwningTenant
+    project: Project
+    displayName: String
+    postfix: String
+    teamChannel: String
```
Comment (kbristow) [inline src/graphql/ingester/OpenShiftProjectEnvironmentCreatedEvent.graphql]: Why is this here? It should be available in the owningTeam. As per our discussion around using the QMToEvent transformers.

Comment (andrejonathan) [reply src/graphql/ingester/OpenShiftProjectEnvironmentCreatedEvent.graphql]: If I use that transformer it has a massive knock on effect to other code that I am not happy to chase right now, pull the branch and implement it and you will see what I mean, let me know, thanks

Comment (kbristow) [reply src/graphql/ingester/OpenShiftProjectEnvironmentCreatedEvent.graphql]: I do not believe it does... This looks like the full change to me...
```
        const openShiftProjectEnvironmentCreatedEvent = {
            owningTeam: GluonToEvent.team(this.owningTeam),
            namespace: environment.namespace,
            masterUrl: this.openshiftEnvironment.masterUrl,
            owningTenant: this.environmentsRequestedEvent.owningTenant,
            project: this.environmentsRequestedEvent.project,
            displayName: environment.displayName,
            postfix: environment.postfix,
        };
```
